src/parser/core/modules/Medicated.ts

```diff
@@ -0,0 +1,71 @@
+import {Events} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {dependency} from '../Injectable'
+import {Data} from './Data'
+import {SimpleRow, StatusItem, Timeline} from './Timeline'
+
+interface Medication {
+	start: number
+	end?: number
+}
+
+export class Medicated extends Analyser {
+	static handle = 'medicated'
+
+	@dependency private data!: Data
+	@dependency private timeline!: Timeline
+
+	private pot: Medication | undefined
+	private pots: Medication[] = []
+
+	initialise() {
+		this.addEventHook({
+			type: 'statusApply',
+			target: this.parser.actor.id,
+			status: this.data.statuses.MEDICATED.id,
+		}, this.onMedication)
+
+		this.addEventHook({
+			type: 'statusRemove',
+			target: this.parser.actor.id,
+			status: this.data.statuses.MEDICATED.id,
+		}, this.offMedication)
+
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onMedication(event: Events['statusApply']) {
+		this.pot = {start: event.timestamp - this.parser.pull.timestamp}
+	}
+
+	private offMedication(event: Events['statusRemove']) {
+		// Pre-pull should handle if we're missing a start time
+		if (!this.pot) {
+			this.pot = {start: this.parser.pull.timestamp}
+		}
```

Comment:
Does this ever trip? Pretty sure the prepull adapter will ensure there's a `statusApply`.

Also, for the _current_ pre-zeroed values, this should be 0, though it's the correct value when storing event timestamps.